craft_application/services/lifecycle.py

```diff
@@ -119,7 +119,8 @@ class LifecycleService(base.ProjectService):
     :param project: The Project object that describes this project.
     :param work_dir: The working directory for parts processing.
     :param cache_dir: The cache directory for parts processing.
-    :param build_for: The architecture or platform we are building for.
+    :param build_plan: The build plan that has already been filtered for the
+        run's platform, build-for and host architectures.
```

Comment:
this is where the mental model for me switches, I would phrase this (with how I wanted it to be perhaps)

```suggestion
    :param build_plan: The filtered build plan of platforms that are valid for
        the running host.
```